docs/developer/checkout/performance.mdx

```diff
@@ -0,0 +1,31 @@
+---
+title: Checkout performance guidelines
+sidebar_label: Performance
+---
+
+This chapter contains guidelines for improving the performance of the checkout process.
+
+## Taxes recalculation with custom taxes apps
+
+When taxes app is used to calculate taxes (`CHECKOUT_CALCULATE_TAXES` AND `ORDER_CALCULATE_TAXES` subscriptions are being used),
+the app is called every time taxes [need to be recalculated](/developer/extending/webhooks/synchronous-events/tax).
+
+You may see mutations like `checkoutLineAdd` or `checkoutShippingMethodUpdate` being slow,
+just like the queries that require prices affected by taxes (see [TaxedMoney](/api-reference/miscellaneous/objects/taxed-money)).
+
+To improve checkout (or order - it can be used in the same way) performance, you can optimize your storefront to avoid recalculating taxes when possible.
+
+Example patterns:
+- When calling `checkoutLineAdd` do not require entire checkout in the response. Check for `errors` and assume mutation succeeds if there are none.
```

Comment:
**Question:** can we tell what fields in mutation can cause slow perf with custom tax app?